lascli/parser/datasets.py

```diff
@@ -19,6 +44,54 @@ def delete_dataset(las_client: Client, dataset_id, delete_documents):
     return las_client.delete_dataset(dataset_id, delete_documents=delete_documents)
 
 
+def sync(
+    las_client: Client,
+    dataset_id,
+    documents,
+    chunk_size,
+    documents_uploaded,
+    documents_failed,
+    num_threads,
+):
+    log_file = Path(documents_uploaded)
+    error_file = Path(documents_failed)
+    documents = json.loads(Path(documents).read_text())
+    uploaded_files = []
+    counter = Counter()
+
+    if log_file.exists():
+        uploaded_files = log_file.read_text().splitlines()
+
+    if error_file.exists():
+        logging.warning(f'{error_file} exists and will be appended to')
+
+    with log_file.open('a') as lf, error_file.open('a') as ef, ThreadPoolExecutor(max_workers=num_threads) as executor:
+        num_docs = len(documents)
+        start_time = time()
+        print(f'start uploading {num_docs} document in chunks of {chunk_size}...')
+
+        for n, chunk in enumerate(group(documents, chunk_size)):
+            fn = partial(_create_documents_pool, client=las_client, dataset_id=dataset_id)
+            inp = [(item, documents[item]) for item in chunk if item is not None and item not in uploaded_files]
+
+            for name, uploaded, reason in executor.map(fn, inp):
+                if uploaded:
+                    lf.write(name + '\n')
+                    counter['uploaded'] += 1
+                    message = f'successfully uploaded {name}'
+                else:
+                    ef.write(name + '\n')
+                    message = f'failed to upload {name}: {reason}'
+                    counter['failed'] += 1
+
+                print(message)
+
+            print(f'{(time() - start_time) / 60:.2f}m: {n * chunk_size}/{num_docs} docs processed '
+                  f'| {n * chunk_size / num_docs * 100:.1f}%')
```

Comment:
Would it be more readable to calculate these numbers before constructing the entire string?